### ISSUE TITLE
Problems to convert CAMx file to Netcdf 

### ISSUE THREAD

**Issue by Ama56** [barronh/pseudonetcdf#91]
Hey, 
I am getting error while converting CAMx output files to Netcdf. Can anyone help to fix this?
Thanks.

I used the following python script 
---------------------------------------------------------------------------------------------------------
#!/usr/bin/env python3
## -- script starts on line above --
import PseudoNetCDF as pnc
from glob import glob
from os.path import join, basename
#
#
inpaths= glob('./Test_New/Output/netcdf/*.grd*')
outroot = './Test_New/Output/netcdf/convert/'
#
for inpath in inpaths:
    outpath = join(outroot, basename(inpath)) + '.nc'
    f = pnc.pncopen(inpath, format='uamiv')
    del f.NAME
    tmp = f.save(outpath)

# -- end script --

--------------------------------------------------------------------------------------------------------------

Errors that I got in below
.....................................................................................................................................................................................................
Traceback (most recent call last):
  File "camx2netcdf.py", line 14, in <module>
    f = pnc.pncopen(inpath, format='uamiv')
  File "/apps2/python/3.6.3/lib/python3.6/site-packages/PseudoNetCDF/_getreader.py", line 77, in pncopen
    outfile = reader(*args, **kwds)
  File "/apps2/python/3.6.3/lib/python3.6/site-packages/PseudoNetCDF/camxfiles/uamiv/Memmap.py", line 98, in __init__
    self.__readheader()
  File "/apps2/python/3.6.3/lib/python3.6/site-packages/PseudoNetCDF/camxfiles/uamiv/Memmap.py", line 246, in __readheader
    data_block_fmt=dtype(dict(names=['DATE']+self.__var_names__, formats=[date_time_fmt]+[spc_3d_fmt]*nspec))
ValueError: name already used as a name or title
......................................................................................................................................................................................................

**Comment by barronh** [comment]
Ama,


1. Can you post a file somewhere that I can download the exact file to test it?
2. Are these standard files or the result of some post-processor?

Thanks,
Thanks,

**Comment by barronh** [comment]
Ama,

I'm glad to help, but I need a test file. You can put one on dropbox and share the link with me.

**Comment by Ama56** [comment]
Hi Barronh,

They are CAMx output files and standard files. 
How can I share my files in dropbox with you?

Thanks.

**Comment by barronh** [comment]
Google “share link Dropbox”   By the way, find just one file that fails. I don’t need them all.

**Comment by Ama56** [comment]
I just have shared a file with you.

Thank you

**Comment by barronh** [comment]
Ama,

Your file has the same variable twice. Ideally, you would just make files without duplicate variables. While duplicating variables has no utility, I am trying to find a way to help without recreating the files.

First, to demonstrate the problem the code below reads and prints the species names section of the header. Note that in the output XO2N is both before and after XOPN.

Code
```
with open('../../Downloads/CAMx.v6.00.continental.12km.MPICH2.20110104.avrg.grd01', 'rb') as gf:
    gf.seek(408)
    keys = gf.read(1320)

print([k.replace(' ', '') for k in keys.decode().split('    ') if k.replace(' ', '') != ''])
```

Output:
```
['CRN2', 'CRNO', 'CRON', 'HNO3', 'HONO', 'N2O5', 'NO', 'NO2', 'NO3', 'NTR', 'OPAN', 'OPEN', 'PAN', 'PANX', 'PNA', 'XO2N', 'XOPN', 'XO2N', 'SO2', 'SULF', 'INTR', 'PNO3', 'PSO4', 'PNH4', 'SOA1', 'SOA2', 'SOA3', 'SOA4', 'SOA5', 'SOA6', 'SOA7', 'SOPA', 'NH3']
```

Now to work around this weird issue. There is an older slower version of the reader, which might help here. It doesn't have all the features of the newer writer, but is a little more flexible. The code below works on my machine.

```
import PseudoNetCDF as pnc
from glob import glob
import os


inpaths = glob('../../Downloads/CAMx.v6.00.continental.12km.MPICH2.????????.avrg.grd01')

for inpath in inpaths:
    outpath = os.path.basename(inpath) + '.nc'
    # Read in the file using the old reader
    f = pnc.pncopen(inpath, format='camxfiles.uamiv.Read.uamiv')

    # Do something with each variable (effecively loads it)
    for k in list(f.variables):
        f.variables[k].dimensions

    # Get rid of the underlying file object.
    del f.rffile
    del f.record_fmt
    del f.cell_count
    del f.padded_size
    del f.record_size
    del f.data_start_byte
    del f.idum
    f.nspec = f.nspec - 1

    # Copy the file to remove the duplicate key
    df = f.copy()

    # Save the file out
    df.save(outpath)
```

These netCDF files will not have the same metadata as the standard uamiv format output. Does this solve your need?

**Comment by Ama56** [comment]
Hey Barron,
I apologize for this. I mistakenly wrote one variable twice in my script. 

I tried with this code that you provided. I got error as below.

........................................................................................................................................................................................................
**PNC:/apps2/python/3.6.3/lib/python3.6/site-packages/PseudoNetCDF/pncgen.py:78:UserWarning:
  Could not add spcnames to file; <class 'OSError'>: array string attributes can only be written with NETCDF4
.........................................................................................................................................................................................................

I think that recreating files will be the easiest solution for me . At least you helped me to figure it out what is creating problems while converting these files. 
Also, you gave your valuable time to help me. I appreciate  it. 

Thanks.
